agent/consul/catalog_endpoint_test.go

```diff
@@ -783,6 +783,20 @@ func TestCatalog_ListNodes(t *testing.T) {
 	if out.Nodes[0].Address != "127.0.0.1" {
 		t.Fatalf("bad: %v", out)
 	}
+
+	t.Run("with option AllowNotModifiedResponse", func(t *testing.T) {
+		args.QueryOptions = structs.QueryOptions{
+			MinQueryIndex:            1,
+			MaxQueryTime:             20 * time.Millisecond,
+			AllowNotModifiedResponse: true,
+		}
+		err := msgpackrpc.CallWithCodec(codec, "Catalog.ListNodes", &args, &out)
+		require.NoError(t, err)
+
+		require.Equal(t, out.Index, uint64(1))
+		require.Len(t, out.Nodes, 0)
+		require.True(t, out.NotModified, "NotModified should be true")
+	})
```

Comment:
Could add an assertion that `NotModiefied` is `false` in the other cases?

Reply:
Done!